src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs

```diff
@@ -187,36 +187,31 @@ public void TestMachineName()
             Assert.NotNull(_process.MachineName);
         }
 
-        [Fact]
+        [Fact, PlatformSpecific(~PlatformID.OSX)]
         public void TestMainModule()
         {
+            // Ensure the process has loaded the modules.
+            while (_process.Modules.Count == 0) ;
+
             // Get MainModule property from a Process object
             ProcessModule mainModule = _process.MainModule;
+            Assert.NotNull(mainModule);
+            Assert.Equal(CoreRunName, Path.GetFileNameWithoutExtension(mainModule.ModuleName));
 
-            if (!global::Interop.IsOSX) // OS X doesn't currently implement modules support
-            {
-                Assert.NotNull(mainModule);
-            }
-
-            if (mainModule != null)
+            // Check that the mainModule is present in the modules list.
+            bool foundMainModule = false;
+            if (_process.Modules != null)
             {
-                Assert.Equal(CoreRunName, Path.GetFileNameWithoutExtension(mainModule.ModuleName));
-
-                // Check that the mainModule is present in the modules list.
-                bool foundMainModule = false;
-                if (_process.Modules != null)
+                foreach (ProcessModule pModule in _process.Modules)
```

Comment:
Could we just do Assert.Contains if we are just trying to see if the item is in this list?